src/components/Menu/MenuItem.tsx

```diff
@@ -0,0 +1,152 @@
+import { useRouter } from 'next/router';
+import { ReactElement, useCallback, useEffect, useState } from 'react';
+import { Link as AmplifyUILink, Flex, View } from '@aws-amplify/ui-react';
+import { IconExternalLink, IconChevron } from '@/components/Icons';
+import Link from 'next/link';
+import { Platform } from '@/data/platforms';
+
+enum Levels {
+  Category = 1,
+  Subcategory = 2,
+  Page = 3
+}
+
+const pathHelper = (path: string) => {
+  const prefix = 'pages/[platform]/';
+
+  const pathNoPrefix = path.slice(prefix.length).split('.')[0];
+
+  const lastIndex = pathNoPrefix.lastIndexOf('/index');
+
+  if (lastIndex > -1) {
+    return pathNoPrefix.substring(0, lastIndex);
+  } else {
+    return pathNoPrefix;
+  }
+};
+
+type PageNode = {
+  title: string;
+  platforms: string[];
+  children: PageNode[];
+  route: string;
+  isExternal: boolean;
+};
+
+type MenuItemProps = {
+  pageNode: PageNode;
+  parentSetOpen: React.Dispatch<React.SetStateAction<boolean>> | null;
+  level: number;
+  currentPlatform: Platform;
+};
+
+export function MenuItem({
+  pageNode,
+  parentSetOpen,
```

Comment:
Here I'm passing that the `setOpen` to the child so that it can call it to have the menu be opened. Basically only "Page" level nodes would need to use this

Is there a better way to do this?